src/main/java/net/minecraftforge/client/event/RenderLevelStageEvent.java

```diff
@@ -0,0 +1,145 @@
+package net.minecraftforge.client.event;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.math.Matrix4f;
+
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.event.IModBusEvent;
+
+/**
+ * Fires at various times during LevelRenderer.renderLevel. 
+ * Check {@link #stage} to render during the appropriate time for your use case.
+ */
+public class RenderLevelStageEvent extends Event
+{
+    public final Stage stage;
+    public final LevelRenderer levelRenderer;
+    public final PoseStack poseStack;
+    public final Matrix4f projectionMatrix;
+    public final int ticks;
```

Comment:
What does `ticks` represent? The name is unclear on this, so it should be clarified in the name or in a javadoc comment.